File: backends/spring-java/src/main/java/dev/stackverse/backend/common/Time.java
package dev.stackverse.backend.common;

import java.time.Instant;
import java.time.temporal.ChronoUnit;

public final class Time {
    private Time() {
    }

    /**
     * PostgreSQL timestamptz stores microseconds; truncating keeps in-memory values
     * identical to a later read, which matters for keyset cursors.
     */
    public static Instant nowUtc() {
        return Instant.now().truncatedTo(ChronoUnit.MICROS);
    }
}

File: backends/spring-kotlin/src/main/kotlin/dev/stackverse/backend/common/Time.kt
package dev.stackverse.backend.common

import java.time.Instant
import java.time.temporal.ChronoUnit

/**
 * PostgreSQL stores timestamps with microsecond precision while [Instant.now] carries
 * nanoseconds. Truncating up front keeps in-memory values identical to what a re-read
 * returns — the v2 keyset cursor compares timestamps and must not be off by nanoseconds.
 */
fun nowUtc(): Instant = Instant.now().truncatedTo(ChronoUnit.MICROS)
